elementum-games/shinobi-chronicles
# Sensei Update

Imgur gallery: https://imgur.com/a/FqdeChM (minor styling tweaks since)
Test cases used:
```
All Players:
	VillageHQ
		- Sensei tab to nav: display sensei detail listing with link to sensei/student profiles, restricted by village and sorted by graduated count
Akademi/Genin:
	Profile
		- Not Student: Display section with link to VillageHQ->Sensei
		- Is Student: Display section with sensei details, link to sensei/student profiles
	VillageHQ
		- Display informational section
			- Has active application: display grid with link to sensei, link to revoke application
		- Actions:
			- Apply: check if already have sensei, rank, if target player is sensei, village, accepting students, error if student roster full or duplicate application, create notification for sensei
			- Cancel Application
			- Leave Sensei
	Members
		- If user is a sensei, same village, accepting applications, link to apply (villageHQ)
		- If user is player's sensei, link to abandon sensei (villageHQ)
	Premium
		- Restrict village change if student
		- Delete active student applications for user
	Level-Up
		- Clear sensei_id from user table
		- Remove user_id from sensei table student_list
		- Increment sensei graduated count
Jonin:
	Profile
		- Is Sensei: Display section with sensei details, link to sensei/student profiles and villageHQ (if student available)
		- Actions:
			- Update specialization (dropdown, Tai/Nin/Gen)
			- Update student message (length 500, html parsed)
			- Update recruitment setting - accept students (checkbox toggle)
			- Update recruitment message (length 100, html parsed)
	VillageHQ
		- Display informational section
			- If not sensei: link to exam
			- If sensei: link to resign -> link to confirm
			- If has active application: display grid with link to student, links to approve/deny
		- Actions:
			- Take Exam: check if already sensei, is Jonin, is >= 75, has 5 jutsu mastered
			- Submit Exam: check if already sensei, is Jonin, is >= 75, has 5 jutsu mastered, has correct answers
			- Resign / Confirm: check if sensei
			- Kick Student
			- Accept Application: clear applications if student count = 3
			- Deny Application
	Members
		- If user is student, link to kick (villageHQ)
	Premium
		- Restrict village change if has students
		- Delete active student applications for user
Moderators:
	Village HQ
		- Swap between village views, can clear any recruitment message
```
